### DISCUSSION TITLE
WindowBuilder Preferences

### DISCUSSION THREAD

**Discussion by C4J** [eclipse/windowbuilder#937] [General]
I notice that if I use the Eclipse - Preferences - WIndowBuilder - Swing - LookAndFeel option and try to do almost anything I get the Apple Spinning Ball and Eclipse locks up. I am trying to set the Metal Look and Feel as the default.

I am using a Mac (Apple Silicon) Macos Sonoma 15.1.1 and Eclipse 2024 - 12 with WindowBuilder 1.18.0

Is there a .pref file I can manually edit to accomplish the same ?

Dave

**Comment by ptziegler** [comment]
The preference you're looking for is `/instance/org.eclipse.wb.swing/P_DEFAULT_LAF=Metal`. You can export your workspace preferences, add this entry and import them again.

I've also created https://github.com/eclipse-windowbuilder/windowbuilder/pull/940, which should make sure that the preview is created within the AWT event dispatcher so alternatively, you could also try out the upcoming nightly build.

**Comment by C4J** [comment]
Outstanding support - thank you very much for this.

Dave
